test/components/routers/test_conditional_router.py

```diff
@@ -40,6 +40,47 @@ def test_invalid_condition_field(self):
         with pytest.raises(ValueError, match="Invalid template"):
             ConditionalRouter(routes)
 
+    def test_invalid_output_template_non_string(self):
+        """
+        ConditionalRouter init raises a ValueError with helpful error message when output is not a string
+        """
+        # output is an int instead of a string template
+        routes = [
+            {
+                "condition": '{{ flag == "double" }}',
+                "output": 2,
+                "output_name": "num_additional_outputs",
+                "output_type": int,
+            }
+        ]
+        with pytest.raises(ValueError) as exc_info:
+            ConditionalRouter(routes)
+        error_message = str(exc_info.value)
+        assert "Invalid template for output" in error_message
+        assert "string" in error_message
+        assert "Jinja2 template" in error_message
+        assert "2" in error_message or "2!" in error_message
```

Comment:
I'm a little confused on this statement. When would `2!` appear in the error message?